Dockerfile

```diff
@@ -202,18 +205,25 @@ RUN \
     # Make ENTRYPOINT alternative script available
     chmod +x "${PHYLUM_VENV}/bin/entrypoint.sh"; \
     #
+    # Create an "alias" for `curl` with secure and common options
+    printf '#!/bin/bash\ncurl --proto "=https" --tlsv1.2 -sSfL "$@"\n' > /usr/bin/curls; \
+    chmod +x /usr/bin/curls; \
```

Comment:
I know this was already present previously so I don't want to complain too much about this, but I'm curious how this came to be.

It says it's an alias, but it writes a bash script to a global executable. That seems somewhat crazy to me and I'm not sure why this isn't just using an actual alias or shell function? It's so much more trouble and I'm not even sure if this is actually cleaned up or stays around for the resulting docker file.

The purpose of it in general also seems questionable. You're effectively hiding options behind a separate name. As a result a reader can't tell anymore which options are actually passed and that the executed command is curl. Is that worth it to omit such a short list of options? Also I'm not sure forcing https and tls v1.2 actually provides any security advantages in practice.

Reply:
Using an alias does not work in a compound Docker `RUN` command like this because [aliases can't be defined and used in the same parsing unit](https://github.com/koalaman/shellcheck/wiki/SC2262). This is the error that happens when using an alias:

```
#21 64.38 + chmod +x /opt/venv/bin/entrypoint.sh
#21 64.40 + alias 'curls=curl --proto "=https" --tlsv1.2 -sSfL'
#21 64.40 + phylum-init -vvv --phylum-release latest --global-install
---TRIMMED---
#21 84.67 WARNING  ✅ Installed Phylum CLI v5.7.3                               cli.py:553
#21 84.79 + curls -o n.sh https://raw.githubusercontent.com/tj/n/master/bin/n
#21 84.79 /bin/bash: line 1: curls: command not found
```

The method chosen was deemed more readable than a shell function, but that is just an opinion. The `/usr/bin/curls` file is removed in the "Final cleanup" section.

The reason for creating the alias was to keep the other nine calls to `curl` short enough to fit on one line (120 character width for this project). The secure options come straight from the [recommended install method for installing Rust using `rustup`](https://www.rust-lang.org/tools/install)

I've changed it to a shell function now since that requires fewer lines.